Movement Rework

nedlemethm opened the pull request:
Currently reworking movement to utilize an input manager and cinemachine, in the hopes of polishing the system and making it smoother.

nedlemethm committed:
Movement Rework

Currently reworking movement to utilize an input manager and cinemachine, in the hopes of polishing the system and making it smoother.

Reallife101 reviewed (approved):
They look fine to me, but I can't tell what the changes are. Approved but please explain the changes!

nedlemethm committed:
Little Changes

Nothing really, just pushing to continue working on laptop

nedlemethm committed:
Merge branch 'main' into Leo_MovementWIP

nedlemethm committed:
Movement Polish

- FOV no longer stutters
- can hold or toggle crouch
- added forces at the start of a wall run to make it feel smoother, wall run still needs to be started parallel to wall to actually work you can't just be jump onto it from the side
- dropped input manager and cinemachine as they would take too long to implement with what is already in place
-bug with enemies addressed, players will now walk over enemies as if they are ground

Reallife101 commented:
Looks good!

nedlemethm committed:
Scroll to switch weapons

yeah

nedlemethm committed:
Merge branch 'main' into Leo_MovementWIP

nedlemethm committed:
Updated Dirty Object

-instead of disabling/enabling meshes, now activates/deactivates game objects
-dirty and clean gameObjects are children to empty gameObject that holds healthbar and prop scripts, while the actual dirtyObject has the health script
-this basically acts as a template of how to do other dirty objects when the models are imported

nedlemethm committed:
Wall Run Bug Fix
